Application class and unit tests

IlliaHryshchuk opened the pull request:
- Completed `Application::run()`
 - Fixed some descriptions in Application
 - Added new fields to `Application class`
 - Added FoundationFunctions (`.cpp` and `.h`)
 - Moved `createFoundation()` and `integrateFoundationIntoModel()` to FoundationFunctions
 - Added `ApplicationTests.h`
 - Added functional tests for Application
 - Fixed unit tests for `createFoundation()` and `integrateFoundationIntoModel()`

IlliaHryshchuk committed:
Completed application class:
 - Completed Application::run();
 - Fixed some descriptions in Application
 - Added FoundationFunctions (.cpp and .h)
 - Moved createFoundation() and integrateFoundationIntoModel() to FoundationFunctions
 - Added unit tests for Application

IlliaHryshchuk committed:
list of fixes:
 - Fixed unit tests for createFoundation() and integrateFoundationIntoModel()
 - Fixed Application tests, now it can be used in any platform

IlliaHryshchuk committed:
Tabulation fix is merged

IlliaHryshchuk committed:
fix after merging

IlliaHryshchuk commented:
The functionality for writing metadata is not completed now, so i left a [commented space](https://github.com/VladGribinchuk/FoundationBuilder/blob/020e57b4a58fb81284c0fc8133bb53e466163843/FoundationBuilder/src/Application.cpp#L62-L63) for it

VladGribinchuk commented on FoundationBuilder/include/Application.h:
```diff
@@ -150,4 +150,13 @@ class Application
 private:
     Application() {}
+
+    std::string inputFilepath; //path to input .STL model
```
Think about making CommandLine object the member of Application class. Thus, you won't need duplicate of all these parameters.

IlliaHryshchuk replied:
i have created `CommandLine cl` field in `Application class`. And now i must announce all parameters inside `Application::run()`? if i announce them inside `Application::init()`, they will be visible only inside `init()` method.

![image](https://user-images.githubusercontent.com/45259941/84596347-cd72bc80-ae65-11ea-8c11-9a3e705806ed.png)

In the screenshot all parameters are initiated in method `run()`, not in `init()`. Is it ok?

VladGribinchuk replied:
I mean, you don't need these local `std::string inputfilepath`, `float height`, etc. duplicates, since you can just use `cl.getValueAs()` or `cl.specified()` in all places, where you need parameter values

IlliaHryshchuk replied:
Ok, I misunderstood firstly

VladGribinchuk commented on UnitTests/MeshTests.h:
```diff
@@ -125,50 +128,80 @@ DEFINE_TEST_CASE(MeshWriteBinary)
+};
+
+
 DEFINE_TEST_CASE(FoundationNormals)
```
Please move all of those test cases to ApplicationTests. So that they will be in one place.

IlliaHryshchuk replied:
Okey

IlliaHryshchuk committed:
list of changes:
 - Some tests are moved from MeshTests.h to ApplicationTests.h
 - Duplicating fields are deleted from Application class
